### ISSUE TITLE
rucss_warmup_resource_fetcher is clogging the database

### ISSUE THREAD

**Issue by vmanthos** [wp-media/wp-rocket#4258]
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version ✅ `3.9.1.1`
- Used the search feature to ensure that the bug hasn’t been reported before ✅

If for whatever reason, the `ResourceFetcherProcess` process gets stuck, numerous `rucss_warmup_resource_fetcher` entries are filling up the database. In the case I examined there were more than 193.000 entries:
![](https://i.vgy.me/4wQeKx.jpg)
 
There were several[ PHP Fatal errors](https://github.com/wp-media/wp-rocket/issues/4257) that popped up in the logs even though Remove Unused CSS was disabled.
 
This is because we'll try to process the batches without checking if the feature is active. It's similar to this:
https://github.com/wp-media/wp-rocket/issues/3680
 
**To Reproduce**
 
I don't have a specific way to do this. Preventing the  `ResourceFetcherProcess` from processing batches could be one.
Initially, the issue came up because of having dynamic resources filenames coming from `mod_pagespeed`.
 
**Expected behavior**
 
Our aim should be to prevent the accumulation of batches of `rucss_warmup_resource_fetcher `.

Even if we do that, there may be cases that we haven't thought of or are beyond our control. We can safeguard this by bailing out from processing the batches if Remove Unused CSS  isn't active. That will give customers a simple way to stop errors.
 
**Additional context**

Clearing those entries can be done using the following SQL command:
```SQL
SELECT * FROM `wp_options` WHERE `option_name` LIKE "rocket_rucss_warmup_resource_fetcher%"
```
 and that will stop any errors coming from that.

**Related issues:** #4207, #4257
**Ticket:** https://secure.helpscout.net/conversation/1585279114/283557?folderId=2135277
 
**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by vmanthos** [comment]
**Probably related:** https://secure.helpscout.net/conversation/1590090544/284922

**Comment by camilamadronero-zz** [comment]
Probably related https://secure.helpscout.net/conversation/1623540338/292771/

**Comment by pgirzalsky** [comment]
Is there any update on this? I experienced the same issue on a client's website. The wp_options table was 52 GB in size because of this issue and made removing the entries and restoring it to a clean state extremely time-consuming because it basically brought the whole server to hold.

**Comment by NataliaDrause** [comment]
Related: https://secure.helpscout.net/conversation/1665639197/301956/

**Comment by NataliaDrause** [comment]
Related: https://secure.helpscout.net/conversation/1672219234/303404/

**Comment by DahmaniAdame** [comment]
Related - https://secure.helpscout.net/conversation/1714525891/312138

**Comment by NataliaDrause** [comment]
Related: https://secure.helpscout.net/conversation/1806895008/329831/

**Comment by piotrbak** [comment]
Closed by 3.11. We are not warming up resources anymore.
